### DISCUSSION TITLE
Error 422: Unprocessable entity when I send form data from JS script

### DISCUSSION THREAD

**Discussion by kareemamrr** [fastapi/fastapi#7038] [Questions]
#### Backend
```py
from fastapi import FastAPI, File, Form, UploadFile, Request
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

origins = ["http://127.0.0.1:5500"]
app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)


@app.post("/test")
def test(username: str = Form(...)):
    print(f"########### Received {username} #############")
```
### Frontend
```js
async function sendData() {
    formData = new FormData();
    formData.append('username', 'Kareem');
    const response = await fetch('http://127.0.0.1:8000/test', {
        method: 'POST',
        data: formData
    })
    console.log(response)
```
### Response
```
Response {
body: (...)
bodyUsed: false
headers: Headers {}
ok: false
redirected: false
status: 422
statusText: "Unprocessable Entity"
type: "cors"
url: "http://127.0.0.1:8000/test"
__proto__: Response }
```
### Description
- The `/test` endpoint is simply supposed to recieve a form containing only `username` and print it out.
- When I test the API using swagger docs it works fine as expected, but when I try to send form data from an external script it throws me this error and that response object above.
### Environment
- OS: Linux
- FastAPI version: 0.61.1
- Python version: 3.8.2

**Comment by ycd** [comment]
Try sending a request with a header.

```javascript
const response = await fetch('http://127.0.0.1:8000/test', {
        method: 'POST',
        data: formData,
        headers: {'content-type': 'multipart/form-data'}
    })
```

**Comment by kareemamrr** [comment]
@ycd the error is now `400 Bad request`

**Comment by Mause** [comment]
What is the error in the response?

**Comment by kareemamrr** [comment]
@Mause 
```
Response {
body: (...)
bodyUsed: false
headers: Headers {}
ok: false
redirected: false
status: 400
statusText: "Bad Request"
type: "cors"
url: "http://127.0.0.1:8000/test"
__proto__: Response }
```

**Comment by Mause** [comment]
Click on the three little dots next to body?

**Comment by kareemamrr** [comment]
@Mause 
```
Response {
body: ReadableStream
    locked: false
    __proto__: ReadableStream
bodyUsed: false
headers: Headers {}
ok: false
redirected: false
status: 400
statusText: "Bad Request"
type: "cors"
url: "http://127.0.0.1:8000/test"
__proto__: Response }
```

**Comment by Mause** [comment]
Try looking at the response in the network tab instead? That will probably be easier

**Comment by kareemamrr** [comment]
@Mause `detail: "There was an error parsing the body"`

**Comment by Mause** [comment]
Is there an error in server logs?

**Comment by ycd** [comment]
> Try sending a request with a header.
> 
> ```js
> const response = await fetch('http://127.0.0.1:8000/test', {
>         method: 'POST',
>         data: formData,
>         headers: {'content-type': 'multipart/form-data'}
>     })
> ```

What about `application/x-www-form-urlencoded`?

**Comment by kareemamrr** [comment]
@Mause Just `INFO:     127.0.0.1:56596 - "POST /test HTTP/1.1" 400 Bad Request`

**Comment by kareemamrr** [comment]
> > Try sending a request with a header.
> > ```js
> > const response = await fetch('http://127.0.0.1:8000/test', {
> >         method: 'POST',
> >         data: formData,
> >         headers: {'content-type': 'multipart/form-data'}
> >     })
> > ```
> 
> What about `application/x-www-form-urlencoded`?

@ycd back to `422 (Unprocessable Entity)`

**Comment by Kludex** [comment]
Usually 400 with `multipart/form-data` are due to the lack of `python-multipart`. Just to be sure, do you have it installed? :pleading_face:

**Comment by kareemamrr** [comment]
@Kludex Yeah I do

**Comment by Mause** [comment]
Which version of python-multipart are you using?

**Comment by kareemamrr** [comment]
@Mause 0.0.5

**Comment by kareemamrr** [comment]
The request went through when using XMLHttpRequest. Closing this issue.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:

**Comment by dash-alex-berndt** [comment]
Has anyone figured out what the issue was? I have a frontend which must use .fetch() so cannot use XMLHttpRequest.

**Comment by NukeMyDay** [comment]
Hello, I ran into the same issue, using fetch and XMLHttpRequest.
@kareemamrr Do you have any explanation what append at your end that caused that 422 error? I'm struggling with that since 2 weeks, without any progress.

**Comment by Saniya-3618** [comment]
I am getting same error using fetch. Anyone have solution?

**Comment by GrahamDE-Santana** [comment]
Do someone have a solution ? I have the same issue
